server/otlp/http_server.go

```diff
@@ -0,0 +1,205 @@
+package otlp
+
+import (
+	"bytes"
+	"compress/gzip"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"strings"
+
+	"github.com/gorilla/handlers"
+	"github.com/gorilla/mux"
+
+	"go.opentelemetry.io/collector/pdata/ptrace/ptraceotlp"
+	pb "go.opentelemetry.io/proto/otlp/collector/trace/v1"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/proto"
+)
+
+const (
+	protoBufContentType = "application/x-protobuf"
+	jsonContentType     = "application/json"
+)
+
+type httpServer struct {
+	addr     string
+	ingester ingester
+
+	hServer *http.Server
+}
+
+func NewHttpServer(addr string, ingester ingester) *httpServer {
+	return &httpServer{
+		addr:     addr,
+		ingester: ingester,
+	}
+}
+
+func (s *httpServer) Start() error {
+	r := mux.NewRouter()
+	r.HandleFunc("/v1/traces", s.Export).Methods("POST")
+
+	s.hServer = &http.Server{
+		Addr:    s.addr,
+		Handler: handlers.CompressHandler(decompressBodyHandler(handlers.ContentTypeHandler(r, protoBufContentType, jsonContentType))),
```

Comment:
it look like you're using 2 separated gzip handlers: the `handlers.CompressHandler` and `decompressBodyHandler`. Am I understanding this correctly?

Reply:
Yes, so the predefined mux compress handler only "compresses" the response, but does not "decompresses" the input, so I had to add a new handler to do that.